### PR TITLE
Add sync docs and update diagram

### PR DESCRIPTION
**Pull request by andreievg**
Links to #2062-4 (add documentation for omSupply central server)

# 👩🏻‍💻 What does this PR do? 

Adds docs and diagram

# 🧪 How has/should this change been tested? 

## 📃 Documentation

_No user facing changes_

or

- [ ] Functional change 1
- [ ] Functional change 2

### CODE REVIEW

File: server/service/src/sync/README.md [hunk 1]
```diff
@@ -84,15 +84,58 @@ Please see below commit for an example of adding a new table and relative transl
 
 [Add translations](https://github.com/msupply-foundation/open-msupply/commit/a9b3d90a597f9670687f0fc26a5632310fe6b4c9)
 
+## Central Servers
+
+In order to use Open mSupply a central server is required, it’s used to configure central data and as a central hub for synchronisation of records from remote sites via the REST API that it hosts.
```
Comment (mark-prins) [inline server/service/src/sync/README.md:89]:
```suggestion
In order to use Open mSupply a central server is required. This is used to configure central data and as a central hub for synchronisation of records from remote sites via the REST API that it hosts. 
```

File: server/service/src/sync/README.md [hunk 2]
```diff
@@ -84,15 +84,58 @@ Please see below commit for an example of adding a new table and relative transl
...
+
+## Open mSupply Central Server
+
+Existing remote sites sync to both Original and Open mSupply at the same time, pushing different records to different endpoints. To reduce the number of configurations on remote site Open mSupply central server is hosted on the same machine as Original mSupply central server, and it exposes the sync API on a known port (Original mSupply port + 2).
+
+An environmental variable flag IS_CENTRAL_SERVER is used to identify an Open mSupply instance as a central server. Open mSupply is still a remote site, from the perspective of Original mSupply central server, and can have active stores. 
+
+When remote sites syncs to Open mSupply’s central server it passes through Original mSupply sync settings, including remote site’s credentials and its own hardware id, Open mSupply central server will use these credentials to check validity of the site against Original mSupply central server.
```
Comment (mark-prins) [inline server/service/src/sync/README.md:101]: :nit 
```suggestion
When a remote site syncs to Open mSupply’s central server it passes through original mSupply sync settings, including the remote site’s credentials and its own hardware id; Open mSupply central server will use these credentials to check validity of the site against the original mSupply central server.
```

File: server/service/src/sync/README.md [hunk 3]
```diff
@@ -84,15 +84,58 @@ Please see below commit for an example of adding a new table and relative transl
...
+
+Open mSupply's central server uses ChangeLog to keep track of what records have been updated in order to figure out what needs to be sent to a remote site. And the remote site keeps track of a cursor (SyncPullCursorV6) to ensure only new updates are synced from the central server.
```
Comment (mark-prins) [inline server/service/src/sync/README.md:103]: :nit
```suggestion
Open mSupply's central server uses ChangeLog to keep track of which records have been updated in order to figure out what needs to be sent to a remote site. And the remote site keeps track of a cursor (SyncPullCursorV6) to ensure only new updates are synced from the central server.
```

File: server/service/src/sync/README.md [hunk 4]
```diff
@@ -84,15 +84,58 @@ Please see below commit for an example of adding a new table and relative transl
...
+
+## ChangeLog instead of queue
+
+In Original mSupply central server, remote/transfer/shared records are added to a sync queue for the related remote site. This queue is used to figure out what should go to what site when there are sync API requests from remote sites.
+
+In Open mSupply, ChangeLog is used for this. The logic, of determining which records should go to which site, happens in one sql statements on ChangeLog, which would look something like this:
```
Comment (mark-prins) [inline server/service/src/sync/README.md:109]:
```suggestion
In Open mSupply, ChangeLog is used for this. The logic, of determining which records should go to which site, happens in one sql statement on the `ChangeLog` table, which would look something like this:
```

**Review by mark-prins — APPROVED** [review]
⭐ this is very exciting! it's a whole new world opening up!

I have some nits to pick - the docs are clear and read well though, which is the main thing

**Comment by andreievg** [comment]
Thanks for checking through the docs and for suggestions !

**Pull request merged**